routes.js

```diff
@@ -1,58 +1,76 @@
-const { error } = require('console');
 const express = require('express');
-const router = express.Router();
+const createMyCustomApi = express.Router();
+const {loadMembers, saveMembers} = require('./dataAccess')
 
-// Middleware
-router.use(express.json());
+// Endpoint to get a list of all team members (get request)
+createMyCustomApi.get('/getAllMembers', function(req, res) {
+    const members = loadMembers();
+    res.status(200).json(members);
+})
 
-// Get method (Get all member)
-router.get('/getAllMembers', (req, res) => {
-  res.send(teamMembersArray);
-  if(!error) {
-      res.status(200);
-  }
+// Endpoint to add a new member to the team (post request)
+createMyCustomApi.post('/addNewMember', (req, res) => {
+    const newMember = {
+        memberId: req.body.memberId,
+        role: req.body.role,
+        firstname: req.body.firstname,
+        lastname: req.body.lastname
+    }
+    if (!newMember.memberId || !newMember.role || !newMember.firstname || !newMember.lastname) {
+        res.status(400).json({ message: 'All fields are required...' });
+        return
+    }
+    const members = loadMembers();
+    if(members.some(member => member.memberId === newMember.memberId)) {
+        res.status(409).json({ message: 'This record ID already exists...' });
+        return
+    }
+    members.push(newMember);
+    saveMembers(members);
+    res.status(200).json({ message: 'OK' });
 });
 
-// Post method (Add a new member)
-router.post('/addNewMember', (req, res) => {
-  const member = {
-    memberId: req.body.memberId,
-    role: req.body.role,
-    firstname: req.body.firstname,
-    lastname: req.body.lastname
-  }
-  teamMembersArray.push(member);
-  res.send(member);
-  if(!error) {
-    res.status(200);
-  }
+// Endpoint to update the information about an existing team member (put request)
+createMyCustomApi.put('/updateExistMember/:memberId', (req, res) => {
+    const memberToUpdateId = parseInt(req.params.memberId, 10);
+    let members = loadMembers();
+    if(!members.some(member => member.memberId === memberToUpdateId)) {
+        res.status(404).json({ message: 'The record with the required ID does not exist...' });
+        return;
+    }
+    const updatedMember = {
+        memberId: req.body.memberId,
+        role: req.body.role,
+        firstname: req.body.firstname,
+        lastname: req.body.lastname
+    }
+    if (!updatedMember.memberId || !updatedMember.role || !updatedMember.firstname || !updatedMember.lastname) {
+        res.status(400).json({ message: 'All fields are required...' });
+        return
+    }
+    const filteredData = members.filter((i) => i.memberId === memberToUpdateId);
+    filteredData.forEach((i) => {
+        i.memberId = updatedMember.memberId,
+        i.role = updatedMember.role,
+        i.firstname = updatedMember.firstname,
+        i.lastname = updatedMember.lastname
+    });
```

Comment:
One of the important principles when writing backend code functionality is making the data and functionality as much immutable as possible. So any developer working with that code should keep in mind minimum amount of state and how that state can be modified.
In simple words - object modification is evil, always prefer creating new object entirely replacing the old one.
Can come up with a more immutable alternative for this code?

Reply:
adjusted